homeassistant/components/playstation_network/media_player.py

```diff
@@ -0,0 +1,146 @@
+
+from enum import StrEnum
+import logging
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityDescription,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PlaystationNetworkCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_MAP = {"PS5": "PlayStation 5", "PS4": "PlayStation 4"}
+
+
+class PlatformType(StrEnum):
+    """PlayStation Platform Enum."""
+
+    PS5 = "PS5"
+    PS4 = "PS4"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Media Player Entity Setup."""
+    coordinator: PlaystationNetworkCoordinator = config_entry.runtime_data
+
+    @callback
+    def add_entities() -> None:
+        if coordinator.data.platform is None:
+            username = coordinator.data.username
+            _LOGGER.warning(
+                "No console found associated with account: %s. -- Pending creation when available",
+                username,
+            )
+            return
+
+        async_add_entities(
+            MediaPlayer(coordinator, platform)
+            for platform in coordinator.data.registered_platforms
+        )
+        remove_listener()
+
+    remove_listener = coordinator.async_add_listener(add_entities)
+    add_entities()
+
+
+class MediaPlayer(CoordinatorEntity[PlaystationNetworkCoordinator], MediaPlayerEntity):
+    """Media player entity representing currently playing game."""
+
+    _attr_media_image_remotely_accessible = True
+    _attr_media_content_type = MediaType.GAME
+
+    def __init__(
+        self, coordinator: PlaystationNetworkCoordinator, platform: str
+    ) -> None:
+        """Initialize PSN MediaPlayer."""
+        super().__init__(coordinator)
+
+        self.entity_description = MediaPlayerEntityDescription(
+            key=platform,
+            translation_key="playstation",
+            device_class=MediaPlayerDeviceClass.RECEIVER,
+            name=self.coordinator.user.online_id,
+            has_entity_name=True,
+        )
+        self._attr_unique_id = (
+            f"{coordinator.config_entry.unique_id}_{self.entity_description.key}"
+        )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.coordinator.user.account_id)},
+            name=self.coordinator.user.online_id,
+            manufacturer="Sony Interactive Entertainment",
+            model="PlayStation Network",
+        )
+
+    @property
+    def name(self) -> str:
+        """Name getter."""
+        return PLATFORM_MAP[self.entity_description.key]
+
```

Comment:
I could be missing something, but doesn't this limit the user to adding a single account? When setup this way, each entity is named after only the device (PlayStation X) but what happens when I add my second PSN account? Now I have two devices that would be named the same. This is of course solvable, but it lead me to think it was the wrong approach. The device in this scenario is the user's PSN account, not the physical hardware. I think this makes sense since we are not able to directly talk to the console. We only get details that pertain to it via the playstation network. All the other sensors that the custom integration provides also aren't console dependent but are associated with the user account.

Reply:
Home Assistant would name it `media_player.playstation_5_2`  , there won't be any conflicts.

Reply:
Agreed, but isn't that "wrong"? In the UI, there is no affordance for the user to differentiate them. They are both named "PlayStation 5" and the entity ID only differs with an _2. Which one is associated with my primary account and which one with my alt account?

Reply:
That's quite usual behaviour, many integrations name entities by their device model, sometimes also manufacturer name prepended. Users can rename entities afterwards however they want.